### PR TITLE
added fibonacci using formula along with test cases

### PR DESCRIPTION
**Pull request by dev-madhurendra**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

I added a code which calculates the nth fibonacci number using formula in a very less time.

- [ ✅ ] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [ ✅ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [ ✅ ] This pull request is all my own work -- I have not plagiarized.
- [ ✅ ] I know that pull requests will not be merged if they fail the automated tests.
- [ ✅ ] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [ ✅ ] All new JavaScript files are placed inside an existing directory.
- [ ✅ ] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [ ✅ ] All new algorithms have a URL in their comments that points to Wikipedia or another similar explanation.
- [ ✅ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by appgurueu — CHANGES_REQUESTED** [review]
Please add this to the existing [`Fibonacci.js`](https://github.com/TheAlgorithms/JavaScript/blob/master/Maths/Fibonacci.js) file (and add the tests to the corresponding test file). Use a proper JSDoc comment. Also, use `each` for the test cases.

The formula doesn't need to be a comment, that's redundant with the implementation (also, the formula in the comment seems to differ from the one you implemented?). You can use `x**y` instead of `Math.pow(x, y)`. Note also that Wikipedia suggests a slightly simpler formula for computation by rounding.

Side note: I find calling this O(1) time/space complexity slightly misleading; this is only (trivially) given because the numbers involved have constant size. If this were to use bigints / bigdecimals, it wouldn't be constant time anymore.

File: Maths/test/Fibonacci.test.js
Comment (appgurueu) [inline Maths/test/Fibonacci.test.js:99]: As said, please use Jest's `each` for these.

File: Maths/Fibonacci.js
Comment (appgurueu) [inline Maths/Fibonacci.js:191]: I'd hoist these three variables out of the function.

File: Maths/Fibonacci.js
Comment (appgurueu) [inline Maths/Fibonacci.js:196]: This variable can be inlined.

**Review by appgurueu — DISMISSED** [review]

**Comment by dev-madhurendra** [comment]
@appgurueu When I'll be able to merge this PR ?

**Comment by appgurueu** [comment]
> @appgurueu When I'll be able to merge this PR ?

It's still waiting on a second review from @raklaptudirm.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]
Sorry for the late review, but consider adding a resource to the formula used here.

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**
